products/platform/documentation-site/research/search-design-doc.md

```diff
@@ -1,15 +1,159 @@
+
+### Logging
+
+Algolia includes analytics, such as ["Most popular searches and results most often returned by the search engine."](https://www.algolia.com/products/analytics/?tab=popular)
+
+### Debugging
+
+The Docker image for the scraper is available on [Docker Hub](https://hub.docker.com/r/algolia/docsearch-scraper). Once we have a config file, we can [run the scraper locally](https://community.algolia.com/docsearch/run-your-own.html#run-the-crawl-from-the-docker-image).
+
+#### Debugging the scraper
+
+#### Debugging the search functionality
+
+We could search for the same term in both the Algolia UI and the landing page to confirm if they both return the same results.
+
+### Caveats
+
+- Access Control: By making the documentation landing page public, it becomes difficult, if not impossible, to allow an unauthenticated user to search documentation from private sources (like a private Github repo). One suggestion is that search results lead to public documentation sources, and we include links to private documentation in the public sources. That way, we can rely on existing access control solutions, while still providing a decent solution. One of the major downsides of that approach is auditing, adding, updating, and maintaining links to/from the public and private sources.
+
+### Security Concerns
+
+If the search input dispatched a request directly to the Algolia API, then there'd be minimal risk of the documentation search negatively affecting other systems. If the request was proxied through our API, then we should be able to rely on any existing DOS protections.
+
+### Privacy Concerns
+
+While there shouldn't be any PII in our public documentation sources, it is possible it exists, and hasn't been discovered yet. Therefore, it is possible that our search results could lead public users to sources that contain PII. If PII is discovered in a repo, we have a [checklist](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/engineering/removing-sensitive-data-from-a-repository.md#checklist-for-removing-pii-in-md-file-from-a-documentation-repo) for how to remove it.
+
+### Open Questions and Risks
+
+- Since the documentation landing page will be public only, we will not be able to search private sources of documentation. If searching private documentation becomes a necessary feature in the future, the proposed implementation will likely not be sufficient.
+- How often should we reindex our documentation sources? For perspective, Algolia's open source crawler runs every 24 hours.
+- What should the URL of the documentation landing page be? Since we have `design.va.gov`, does something like `docs.va.gov` make sense?
+- What is the current opportunity cost of building a custom solution? What are we choosing to delay right now by building a custom solution?
```

Comment:
wait, isn't this design doc about the search MVP? and we're using algolia? what custom solution are we talking about here?

Reply:
@billfienberg Correct me if I'm wrong, but I think this is here because it'll take a lot of our own infrastructure and wiring to make this particular solution (Algolia) work. The indexing and whatnot isn't the custom bit, but it's not exactly an out-of-the-box solution since [we can't use their scraper](https://github.com/department-of-veterans-affairs/va.gov-team/pull/5248#discussion_r373544321).

Reply:
@cvalarida, you are correct. We could use Algolia's hosted search engine to avoid maintaining a DB, and maintaining an API to access that DB. We could use Algolia's open source scraper to avoid writing our own scraper. And we could use Algolia's UI library to avoid writing a custom search result renderer. 

But we'd still need to iterate on the configs for the search index, scraper, and UI library, in addition to configuring when to run the scraper (for example, cron job every 24 hours). And I suspect we'd need to review the source and image for the scraper before adding it one of our pipelines. 

TLDR: There are several components to the system, and each one can be an independent buy-or-build decision. 

- Data store
- Way to get data into the data store (scraper)
- Way to get data out of the data store (search API)
- Way to display data to the user

[Algolia's Search API](https://www.algolia.com/products/search/) would satisfy the data store and API components. 

[Algolia's scraper](https://hub.docker.com/r/algolia/docsearch-scraper) would provide a way to get data into the data store. 

And [Algolia's DocSearch UI library](https://github.com/algolia/docsearch) would provide a way to display data to the user.

Reply:
Are we evaluating these things independently? i.e. would we consider using a non-Algolia scraper with Algolia's data store? Or Algolia's scraper with our own data store?

Reply:
We are not evaluating these things independently. I think there'd be little to no value in using Algolia's scraper or JS library without using their index.